Dashboard/app/js/lib/views/devices/assignment-edit-views.jsx

```diff
@@ -357,7 +357,9 @@ FLOW.AssignmentEditView = FLOW.ReactComponentView.extend(
             this.deviceGroups[dgId] = {
               0: {
                 name: 'Select all devices',
-                checked: numberOfDevices === numberOfSelectedDevices,
+                checked:
+                  numberOfDevices !== 0 &&
+                  numberOfDevices === numberOfSelectedDevices,
```

Comment:
Wouldn't the second condition by itself suffice?

Reply:
No, the first to make sure the select-box isn't checked when they're no devices in the group